sdk/azidentity/client_secret_credential.go

```diff
@@ -54,48 +54,65 @@ func (c *ClientSecretCredential) AuthenticationPolicy(options azcore.Authenticat
 }
 
 type bearerTokenPolicy struct {
-	// take lock when manipulating header/expiresOn fields
-	lock      sync.RWMutex
-	header    string
-	expiresOn time.Time
+	empty     chan bool    // indicates we don't have a token yet
+	updating  atomic.Int64 // atomically set to 1 to indicate a refresh is in progress
+	header    atomic.String
+	expiresOn atomic.Time
 	creds     azcore.TokenCredential // R/O
 	scopes    []string               // R/O
 }
 
 func newBearerTokenPolicy(creds azcore.TokenCredential, scopes []string) *bearerTokenPolicy {
-	// set the token as expired so first call to Do() refreshes it
-	return &bearerTokenPolicy{creds: creds, scopes: scopes, expiresOn: time.Now().UTC()}
+	bt := bearerTokenPolicy{creds: creds, scopes: scopes}
+	// prime channel indicating there is no token
+	bt.empty = make(chan bool, 1)
+	bt.empty <- true
+	return &bt
 }
 
 func (b *bearerTokenPolicy) Do(ctx context.Context, req *azcore.Request) (*azcore.Response, error) {
-	var bt string
-	// take read lock and check if the token has expired
-	b.lock.RLock()
+	// check if the token is empty
+	if <-b.empty {
```

Comment:
I'd add some more comments here for other developers. Something like:
// This return true if this is the first goroutine to try to use this token; it returns false to all other goroutines after the 1st token has been obtained.